distribution/lib/Standard/Base/0.0.0-dev/src/Data/Filter_Condition.enso

```diff
@@ -192,10 +193,12 @@ type Filter_Condition
         Not_Like sql_pattern ->
             regex = sql_like_to_regex sql_pattern
             elem -> regex.matches elem . not
-        ## TODO once we have proper hashing we could create a hashmap and
-           answer quicker, currently we need to do a full scan for each element.
-        Is_In values -> values.contains
-        Not_In values -> elem -> values.contains elem . not
+        Is_In values ->
+            set = Set.from_vector values
+            set.contains
```

Comment:
shouldn't be the implementation of `contains` within `Vector`?

Reply:
There is one, but using set will be faster.

This way here we construct the Set (O(N)) operation once, and save that set in the closure of the generated function, so then we can use the quick (~O(1), hashing-based) set checks.